Provision haraka fails

**Describe the bug**
Haraka fails

**To Reproduce**
provision haraka

**Expected behavior**
new haraka jail ispromoted

**Server (please complete the following information):**
 - OS: FreeBSD
 - Version 13.2-RELEASE-p4

**Additional context**
error message:
<snip>
Message from node16-16.20.1_1:

--
Note: If you need npm (Node Package Manager), please install www/npm.
--
===>   NOTICE:

This port is deprecated; you may wish to reconsider installing it:

Upstream EOL reaches on 2023-09-11.

It is scheduled to be removed on or after 2023-09-30.
=====
Message from git-lite-2.42.0:

--
If you installed the GITWEB option please follow these instructions:

In the directory /usr/local/share/examples/git/gitweb you can find all files to
make gitweb work as a public repository on the web.

All you have to do to make gitweb work is:
1) Please be sure you're able to execute CGI scripts in
   /usr/local/share/examples/git/gitweb.
2) Set the GITWEB_CONFIG variable in your webserver's config to
   /usr/local/etc/git/gitweb.conf. This variable is passed to gitweb.cgi.
3) Restart server.


If you installed the CONTRIB option please note that the scripts are
installed in /usr/local/share/git-core/contrib. Some of them require
other ports to be installed (perl, python, etc), which you may need to
install manually.
jexec stage ln -s /usr/local/bin/python3 /usr/local/bin/python
jexec stage npm install -g --omit=dev node-gyp
npm notice 
npm notice New major version of npm available! 9.9.0 -> 10.2.1
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.2.1
npm notice Run npm install -g npm@10.2.1 to update!
npm notice 
npm ERR! code EEXIST
npm ERR! path /usr/local/bin/node-gyp
npm ERR! EEXIST: file already exists
npm ERR! File exists: /usr/local/bin/node-gyp
npm ERR! Remove the existing file and try again, or run npm
npm ERR! with --force to overwrite files recklessly.

npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2023-10-21T19_35_39_314Z-debug-0.log
</snip

End of debug log. I can provide entire log if necessary.
<snip>
 918 timing reify:rollback:retireShallow Completed in 0ms
919 timing command:install Completed in 3144ms
920 notice
920 notice New ^[[31mmajor^[[39m version of npm available! ^[[31m9.9.0^[[39m -> ^[[32m10.2.1^[[39m
920 notice Changelog: ^[[36mhttps://github.com/npm/cli/releases/tag/v10.2.1^[[39m    
920 notice Run ^[[32mnpm install -g npm@10.2.1^[[39m to update!
920 notice
921 verbose stack Error: EEXIST: file already exists
921 verbose stack     at failEEXIST (/usr/local/lib/node_modules/npm/node_modules/bin-links/lib/check-bin.js:47:32)
921 verbose stack     at checkLink (/usr/local/lib/node_modules/npm/node_modules/bin-links/lib/check-bin.js:38:12)
921 verbose stack     at async Promise.all (index 0)
921 verbose stack     at async Function.checkBins (/usr/local/lib/node_modules/npm/node_modules/bin-links/lib/check-bins.js:15:3)
921 verbose stack     at async Arborist.[checkBins] (/usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/rebuild.js:256:5)
921 verbose stack     at async /usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/reify.js:643:7
922 verbose cwd /
923 verbose FreeBSD 13.2-RELEASE-p4
924 verbose node v16.20.1
925 verbose npm  v9.9.0
926 error code EEXIST
927 error path /usr/local/bin/node-gyp
928 error EEXIST: file already exists
929 error File exists: /usr/local/bin/node-gyp
930 error Remove the existing file and try again, or run npm
931 error with --force to overwrite files recklessly.
932 verbose exit 1
933 timing npm Completed in 3298ms      
934 verbose unfinished npm timer reify 1697916939468
935 verbose unfinished npm timer reify:unpack 1697916941953
936 verbose unfinished npm timer reifyNode:node_modules/node-gyp 1697916941953
937 verbose code 1
<snip>

--- a/mail-toaster.sh
+++ b/mail-toaster.sh
@@ -922,7 +922,12 @@ unmount_pkg_cache()
 
 freebsd_release_url_base()
 {
-	echo "$FBSD_MIRROR/pub/FreeBSD/releases/$(uname -m)/$FBSD_REL_VER"
+	_major_ver="$(/bin/freebsd-version | cut -f1 -d.)"
+	if [ "$_major_ver" -lt "13" ]; then
+		echo "http://ftp-archive.freebsd.org/pub/FreeBSD-Archive/old-releases"
+	else
+		echo "ftp://ftp.freebsd.org/pub/FreeBSD/releases"
+	fi
 }
 
 stage_fbsd_package()
@@ -930,8 +935,9 @@ stage_fbsd_package()
 	local _dest="$2"
 	if [ -z "$_dest" ]; then _dest="$STAGE_MNT"; fi
 
-	tell_status "downloading $(freebsd_release_url_base)/$1.txz"
-	fetch -m "$(freebsd_release_url_base)/$1.txz" || exit
+	_file_uri="$(freebsd_release_url_base)/$(uname -m)/$FBSD_REL_VER/$1.txz"
+	tell_status "downloading $_file_uri"
+	fetch -m "$_file_uri" || exit
 	echo "done"
 
 	tell_status "extracting FreeBSD package $1.tgz to $_dest"

--- a/provision/base.sh
+++ b/provision/base.sh
@@ -45,7 +45,7 @@ install_freebsd()
 
 	if [ -n "$USE_BSDINSTALL" ]; then
 		export BSDINSTALL_DISTSITE;
-		BSDINSTALL_DISTSITE="$FBSD_MIRROR/pub/FreeBSD/releases/$(uname -m)/$(uname -m)/$FBSD_REL_VER"
+		BSDINSTALL_DISTSITE="$(freebsd_release_url_base)/$(uname -m)/$(uname -m)/$FBSD_REL_VER"
 		bsdinstall jail "$BASE_MNT"
 	else
 		stage_fbsd_package base "$BASE_MNT"

--- a/provision/dns.sh
+++ b/provision/dns.sh
@@ -176,21 +176,13 @@ test_unbound()
 
 switch_host_resolver()
 {
-	if [ ! -f /etc/resolv.conf.orig ]; then
-		cp /etc/resolv.conf /etc/resolv.conf.orig
-	fi
+	if grep "^nameserver $(get_jail_ip dns)" /etc/resolv.conf; then return; fi
 
-	if ! grep "^nameserver $(get_jail_ip dns)" /etc/resolv.conf;
-	then
-		echo "switching host resolver to dns jail"
-		echo "nameserver $(get_jail_ip dns)
+	echo "switching host resolver to dns jail"
+	sysrc -f /etc/resolvconf.conf name_servers="$(get_jail_ip dns) $(get_jail_ip6 dns)"
+	echo "nameserver $(get_jail_ip dns)
 nameserver $(get_jail_ip6 dns)" | resolvconf -a "$PUBLIC_NIC"
-		sysrc -f /etc/resolvconf.conf name_servers="\"$(get_jail_ip dns) $(get_jail_ip6 dns)\""
-		sysrc -f /etc/resolvconf.conf resolvconf=NO
-		# echo "nameserver $(get_jail_ip dns)" > /etc/resolv.conf
-		# echo "nameserver $(get_jail_ip6 dns)" >> /etc/resolv.conf
-		# cat /etc/resolv.conf.orig >> /etc/resolv.conf
-	fi
+	sysrc -f /etc/resolvconf.conf resolvconf=NO
 }
 
 base_snapshot_exists || exit

--- a/provision/dns.sh
+++ b/provision/dns.sh
@@ -174,6 +174,25 @@ test_unbound()
 	echo "it worked."
 }
 
+switch_host_resolver()
+{
+	if [ ! -f /etc/resolv.conf.orig ]; then
+		cp /etc/resolv.conf /etc/resolv.conf.orig
+	fi
+
+	if ! grep "^nameserver $(get_jail_ip dns)" /etc/resolv.conf;
+	then
+		echo "switching host resolver to dns jail"
+		echo "nameserver $(get_jail_ip dns)
+nameserver $(get_jail_ip6 dns)" | resolvconf -a "$PUBLIC_NIC"
+		sysrc -f /etc/resolvconf.conf name_servers="\"$(get_jail_ip dns) $(get_jail_ip6 dns)\""
+		sysrc -f /etc/resolvconf.conf resolvconf=NO
+		# echo "nameserver $(get_jail_ip dns)" > /etc/resolv.conf
+		# echo "nameserver $(get_jail_ip6 dns)" >> /etc/resolv.conf
+		# cat /etc/resolv.conf.orig >> /etc/resolv.conf
+	fi
+}
+
 base_snapshot_exists || exit
 create_staged_fs dns
 start_staged_jail dns
@@ -182,15 +201,4 @@ configure_unbound
 start_unbound
 test_unbound
 promote_staged_jail dns
-
-if [ ! -f /etc/resolv.conf.orig ]; then
-	cp /etc/resolv.conf /etc/resolv.conf.orig
-fi
-
-if ! grep "^nameserver $(get_jail_ip dns)" /etc/resolv.conf;
-then
-	echo "switching host resolver to $(get_jail_ip dns)"
-	echo "nameserver $(get_jail_ip dns)" > /etc/resolv.conf
-	echo "nameserver $(get_jail_ip6 dns)" >> /etc/resolv.conf
-	cat /etc/resolv.conf.orig >> /etc/resolv.conf
-fi
+switch_host_resolver

--- a/provision/dns.sh
+++ b/provision/dns.sh
@@ -81,10 +81,13 @@ tweak_unbound_conf()
 	sed -i.bak \
 		-e 's/# interface: 192.0.2.153$/interface: 0.0.0.0/' \
 		-e 's/# interface: 192.0.2.154$/interface: ::0/' \
-		-e '/# use-syslog/      s/# //' \
-		-e '/# chroot: /        s/# //; s/".*"/""/' \
-		-e '/# hide-identity: / s/# //; s/no/yes/' \
-		-e '/# hide-version: /  s/# //; s/no/yes/' \
+		-e '/# use-syslog/s/# //' \
+		-e '/# chroot: /s/# //' \
+		-e '/chroot: /s/".*"/""/' \
+		-e '/# hide-identity: /s/# //' \
+		-e '/hide-identity: /s/no/yes/' \
+		-e '/# hide-version: /s/# //' \
+		-e '/hide-version: /s/no/yes/' \
 		-e '/# access-control: ::ffff:127.*/ a\ 
 include: "/data/access.conf" \
 ' \

--- a/provision/haraka.sh
+++ b/provision/haraka.sh
@@ -12,14 +12,14 @@ HARAKA_CONF="$ZFS_DATA_MNT/haraka/config"
 install_haraka()
 {
 	tell_status "installing node & npm"
-	stage_pkg_install npm-node16 gmake python3 pkgconf git-lite || exit
+	stage_pkg_install npm-node18 gmake pkgconf git-lite || exit
 	if [ "$BOURNE_SHELL" != "bash" ]; then
 		tell_status "Install bash since not in base"
 		stage_pkg_install bash || exit
 	fi
-	export PYTHON=/usr/local/bin/python3
-	stage_exec ln -s /usr/local/bin/python3 /usr/local/bin/python
-	stage_exec npm install -g --omit=dev node-gyp || exit
+	# export PYTHON=/usr/local/bin/python3
+	# stage_exec ln -s /usr/local/bin/python3 /usr/local/bin/python
+	# stage_exec npm install -g --omit=dev node-gyp || exit
 
 	# Workaround for NPM bug https://github.com/npm/cli/issues/2610
 	stage_exec bash -c 'git config --global url."https://github.com/".insteadOf git@github.com:'

